fdb-extensions/src/main/java/com/apple/foundationdb/async/hnsw/HNSW.java

```diff
@@ -997,28 +992,23 @@ public CompletableFuture<Void> insert(@Nonnull final Transaction transaction, @N
         return forEach(nodeReferenceAndNodes,
                 nodeReferenceAndNode -> {
                     final AbstractNode<N> node = nodeReferenceAndNode.getNode();
-                    if (node.getNeighbors().isEmpty()) {
-                        final NodeReferenceWithVector nodeReference = nodeReferenceAndNode.getNodeReference();
-                        return fetchBaseNode(readTransaction, storageTransform, nodeReference.getPrimaryKey())
-                                .thenApply(baseCompactNode -> {
-                                    if (baseCompactNode == null) {
-                                        return null;
-                                    }
+                    final NodeReferenceWithVector nodeReference = nodeReferenceAndNode.getNodeReference();
+                    return fetchBaseNode(readTransaction, storageTransform, nodeReference.getPrimaryKey())
```

Comment:
Just to understand the change here in the most recent commit: we're no longer short-circuiting our existence finding of a reference based on whether the node has neighbors, and that's to handle the case that we've deleted and re-created the same node (in an inlining layer). And we're only doing that while doing repairs (not during greedy search for reads, where we still short-circuit if the node has neighbors)

That's probably a fair enough compromise, to be more persnickety about removing recreated nodes when doing repairs